client/state/jetpack-settings/modules/reducer.js

```diff
@@ -23,24 +47,10 @@ import {
  * @param  {Object} action action
  * @return {Array}         Updated state
  */
-export const items = ( state = {}, { type, siteId, moduleSlug } ) => {
-	switch ( type ) {
-		case JETPACK_MODULE_ACTIVATE_SUCCESS:
-			return merge( {}, state, {
-				[ siteId ]: {
-					[ moduleSlug ]: {
-						active: true
-					}
-				}
-			} );
-		// return initial state when serializing/deserializing
-		case SERIALIZE:
-		case DESERIALIZE:
-			return {};
-		default:
-			return state;
-	}
-};
+export const items = createReducer( {}, {
+	[ JETPACK_MODULE_ACTIVATE_SUCCESS ]: createItemsReducer( true ),
```

Comment:
Little concern here. By the way props from the `action` parameter are pulled, this reducer will only be useful for the actions handling a `.moduleSlug` prop. This isn't the case for an action with fetches the list of modules from the API. Should we rename dice `createItemsReducer()` function to something specific to these actions it's able to handle or just write the arguments to the reducer without _destructuring_ ?